### ISSUE TITLE
S109: FP with auto-property assignment

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonar-dotnet#2431]
### Description

Rule [S109](https://jira.sonarsource.com/browse/RSPEC-109): no magic numbers

```
namespace S109
{
    public class Class1
    {
        private int _b = 2; // S109 not raised

        public int B
        {
            get { return _b; }
            set { _b = value; }
        }

        public int A { get; set; } = 2; // S109 raised
    }
}
```

The rule doesn't raise for the long-hand version of the code, but does for the auto-implemented property. The results should be consistent.

Note: this rule is not part of SonarWay -> lower priority.

TODO: does this also need to be fixed for VB.NET.

### Known workarounds

None

### Related information

* SonarC# Version: 7.13
